src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -457,6 +462,127 @@ public RealJenkinsRule prepareHomeLazily(boolean prepareHomeLazily) {
+        String message = "Path " + fipsLibrariesPath.toString() + " must contains the FIPS BouncyCastle jars:";
+        message += "bc-fips.jar, bctls-fips.jar and bcpkix-fips.jar" + System.lineSeparator();
+        message += "Configure your pom.xml with this plugin execution: " + System.lineSeparator();
+        message += "<plugin>\n" +
+                "  <groupId>org.apache.maven.plugins</groupId>\n" +
+                "  <artifactId>maven-dependency-plugin</artifactId>\n" +
+                "  <executions>\n" +
+                "    <execution>\n" +
+                "      <id>copy-libs</id>\n" +
+                "      <goals>\n" +
+                "        <goal>copy</goal>\n" +
+                "      </goals>\n" +
+                "      <configuration>\n" +
+                "        <artifactItems>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bc-fips</artifactId>\n" +
+                "            <version>1.0.2.4</version>\n" +
+                "          </artifactItem>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bcpkix-fips</artifactId>\n" +
+                "            <version>1.0.7</version>\n" +
+                "          </artifactItem>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bctls-fips</artifactId>\n" +
+                "            <version>1.0.17</version>\n" +
+                "          </artifactItem>\n" +
+                "        </artifactItems>\n" +
+                "        <stripVersion>true</stripVersion>\n" +
+                "      </configuration>\n" +
+                "    </execution>\n" +
+                "  </executions>\n" +
+                "</plugin>";
+        return new IllegalArgumentException(message);
+    }
+
+    private Path writeFIPSJavaSecurityFile() throws IOException {
+        String javaHome = System.getProperty("java.home");
+        if(javaHome == null) {
+            throw new IllegalArgumentException("Cannot find java.home property");
+        }
+        Path javaSecurity = Paths.get(javaHome, "conf", "security", "java.security");
+        Properties properties = new Properties();
+        Path securityFile = Files.createTempFile("java", ".security");
+        securityFile.toFile().deleteOnExit();
+        try (InputStream inputStream = Files.newInputStream(javaSecurity);
+             OutputStream outputStream = Files.newOutputStream(securityFile)) {
+            properties.load(inputStream);
+            properties.keySet().removeIf(o -> ((String)o).startsWith("security.provider"));
+            properties.put("security.provider.1", "org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider C:HYBRID;ENABLE{All};");
```

Comment:
Note for others, the hybrid is a psedo random number that is initially seeded from a strong instance and is regularly reseeded from that instance to maintain strength and prevent any predictability.
This should ensure we do not run out of entropy on VMs, however as it does need the initial seed and some subsequent reseeding that is not gauranteed.  (Depending on how often the seeds secure random's source is required.)
in testing a product deployed with this I have not noticed any delays, but if we start seeing timeouts and the SR in the stack trace we would need the infra team to deploy something like [`haveged`](https://linux.die.net/man/8/haveged) on the nodes (both the VM and k8s node pool nodes).  (/cc @dduportal)

Reply:
It's worth adding it (I'll open an issue on jenkins-infra/helpdesk) right now on the agent templates (in https://github.com/jenkins-infra/packer-images/blob/main/provisioning/ubuntu-provision.sh). For VM agents, the service will be started on boot of each agent to help.
We already have the RNGD tools on the non-agent VMs (controllers, web services): https://github.com/jenkins-infra/jenkins-infra/blob/production/dist/profile/manifests/rngd.pp.

For the k8s node pool I have no idea if it is already present or not.
- We are using AKS with managed nodes with Azure Linux OS on the node: https://github.com/jenkins-infra/azure/blob/47e44a47bcd5330dbdffc443eb4b2f5a0c4b398a/ci.jenkins.io-kubernetes-agents.tf#L102-L134
- There is an (old: 2018) issue for AKS about this: https://github.com/Azure/AKS/issues/219 but it's unclear if it was fixed or not.
- On the upstream we have this issue: https://github.com/kubernetes/kubernetes/issues/60751 but again unclear

=> I believe we might need to search for a daemonset-based installation to ensure it's loaded.

Reply:
@jtnord is the "entropy increase" need to be set up (somehow) on Windows agents? (I do not know this platform well enough)

Reply:
> @jtnord is the "entropy increase" need to be set up (somehow) on Windows agents? (I do not know this platform well enough)

JDK on windows uses [`CryptGenRandom`](https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-cryptgenrandom) despite it being deprecated 😱 
the documentation here does talk about the sources and this includes process related things as well as keyboard/mouse.

The new alternative (AFAICT) is [`ProcessPrng`](https://learn.microsoft.com/en-us/windows/win32/seccng/processprng) which is lacking any and all details.
That said windows requirements these days are for TPM which contains a teue random number generator, so I don't think either of these would block like /dev/random.  (and yes there are various TPMs that exists whose TRNG is not actually random!).

also here is a [random Register article](https://www.theregister.com/2007/11/13/windows_random_number_gen_flawed/) saying windows is flawed.  
(which also implies at least in the windows 2000 era, that this was a PRNG not a TRNG...)